src/handler/get_extended_pubkey.c

```diff
@@ -141,27 +142,41 @@ void handler_get_extended_pubkey(dispatcher_context_t *dc, uint8_t protocol_vers
         return;
     }
 
-    char serialized_pubkey_str[MAX_SERIALIZED_PUBKEY_LENGTH + 1];
+    serialized_extended_pubkey_check_t pubkey_check;
+    if (0 > get_extended_pubkey_at_path(bip32_path,
+                                        bip32_path_len,
+                                        BIP32_PUBKEY_VERSION,
+                                        &pubkey_check.serialized_extended_pubkey)) {
+        PRINTF("Failed getting bip32 pubkey\n");
+        SEND_SW(dc, SW_BAD_STATE);
+        return;
+    }
+
+    crypto_get_checksum((uint8_t *) &pubkey_check.serialized_extended_pubkey,
+                        sizeof(pubkey_check.serialized_extended_pubkey),
+                        pubkey_check.checksum);
 
-    int serialized_pubkey_len = get_serialized_extended_pubkey_at_path(bip32_path,
-                                                                       bip32_path_len,
-                                                                       BIP32_PUBKEY_VERSION,
-                                                                       serialized_pubkey_str,
-                                                                       NULL);
-    if (serialized_pubkey_len == -1) {
+    char pubkey_str[MAX_SERIALIZED_PUBKEY_LENGTH + 1];
+    int pubkey_str_len = base58_encode((uint8_t *) &pubkey_check,
+                                       sizeof(pubkey_check),
+                                       pubkey_str,
+                                       sizeof(pubkey_str));
+    if (pubkey_str_len != 111 && pubkey_str_len != 112) {
+        PRINTF("Failed encoding base58 pubkey\n");
         SEND_SW(dc, SW_BAD_STATE);
         return;
     }
+    pubkey_str[pubkey_str_len] = 0;
```

Comment:
For clarity I would have replaced this 0 with `\0`

Reply:
I always did `= 0` in the codebase, so leaving the same for consistency; can be changed everywhere in a separate PR if desired.